pulsar-transaction/buffer/src/main/java/org/apache/pulsar/transaction/buffer/TransactionBufferProvider.java

```diff
@@ -51,6 +52,21 @@ static TransactionBufferProvider newProvider(String providerClassName) throws IO
         }
     }
 
+    static TransactionBufferProvider newProvider(String providerClassName, Object... args) throws IOException {
```

Comment:
this change doesn't make sense to me. it is a generic reflection. the constructor should be consistent across different implementations. if you want the provider to provide the transaction buffer per topic, you should add the `topic` variable to `newTransactionBuffer` method.

```
interface TransactionBufferProvider {
    public CompletableFuture<TransactionBuffer> newTransactionBuffer(String topic);
}
```